sklearn/utils/testing.py

```diff
@@ -697,5 +706,34 @@ def check_skip_travis():
     if os.environ.get('TRAVIS') == "true":
         raise SkipTest("This test needs to be skipped on Travis")
 
+
+def delete_folder(folder_path):
+    """Utility function to cleanup a temporary folder if still existing. Copy from joblib.pool (for independance)"""
+    if os.path.exists(folder_path):
+        shutil.rmtree(folder_path)
+
+
+class TempMemmap(object):
+    def __init__(self, X, mmap_mode='r'):
+        self.temp_folder = tempfile.mkdtemp(prefix='sklearn_testing_')
+        self.mmap_mode = mmap_mode
+        self.X = X
+
+    def __enter__(self):
+        fpath = op.join(self.temp_folder, 'data.pkl')
+        joblib.dump(self.X, fpath)
+        X_read_only = joblib.load(fpath, mmap_mode=self.mmap_mode)
+        atexit.register(lambda: delete_folder(self.temp_folder))
+        return X_read_only
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._clean()
+
+    def _clean(self):
+        try:
+            delete_folder(self.temp_folder)
+        except shutil.WindowsError:
+            warnings.warn("Could not delete temporary folder %s" % self.temp_folder)
```

Comment:
Please remove this `try / except` block once moved into the `delete_folder` function itself.